### PR TITLE
add wafv2 modules

### PR DESCRIPTION
**Pull request by markuman**
##### SUMMARY
Add wafv2 modules

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME

  - wafv2_resources
  - wafv2_resources_info
  - wafv2_rule_group
  - wafv2_rule_group_info
  - wafv2_web_acl
  - wafv2_web_acl_info

##### ADDITIONAL INFORMATION

The modules share same `module_utils/wafv2.py` codebase.  
Furthermore, the integration test needs an Application Loadbalancer, that needs itself needs a target group, a subnet and a VPC. This is the reason why the integration test is that long.

### CODE REVIEW

**Comment by markuman** [comment]
aws terminator needs rw permissions for wafv2

` botocore.exceptions.ClientError: An error occurred (AccessDeniedException) when calling the ListRuleGroups operation: User: arn:aws:sts::966509639900:assumed-role/ansible-core-ci-test-prod/prod=shippable=ansible-collections=community.aws=1704.22 is not authorized to perform: wafv2:ListRuleGroups on resource: arn:aws:wafv2:us-east-1:966509639900:regional/rulegroup/*`

**Comment by markuman** [comment]
Required: https://github.com/ansible/ansible/pull/73975 for pass 2.9 integration tests

**Comment by markuman** [comment]
Ok, shippable is running to many tests in parallel.

Here the wafv2 test pass  
https://app.shippable.com/github/ansible-collections/community.aws/runs/2238/20/console  
Afterwards the wafv2_ip_set set failed (_already in main_) due time exceeded (45min).

I guess someone must kick one single failed test, until all pass.

File: plugins/modules/wafv2_resources.py
Comment (jillr) [inline plugins/modules/wafv2_resources.py:29]:
```suggestion
        - The name of the web acl.
```

File: plugins/modules/wafv2_resources_info.py
Comment (jillr) [inline plugins/modules/wafv2_resources_info.py:79]: Won't this module need to support a NextMarker parameter?

**Review by jillr — COMMENTED** [review]
partial review on the first few files to get you some feedback, will continue reviewing tomorrow

File: plugins/modules/wafv2_resources_info.py
Comment (markuman) [reply plugins/modules/wafv2_resources_info.py]: oh, right!

**Review by jillr — COMMENTED** [review]
Thanks for your patience @markuman (and all the new modules!). If possible I'd like to get a second review on this, maybe @tremble or @goneri?  Just a few docs nits and test comments left from me.

I'm planning to release 1.5 in the next 1-2 weeks, and I'm happy block that on this one as we're really close.

File: plugins/module_utils/wafv2.py
Comment (goneri) [inline plugins/module_utils/wafv2.py:29]: I'm a bit annoyed by the similarity with `wafv2_list_web_acls` but it sounds like we cannot really merge both without creating something worst :-).

Comment (markuman) [reply plugins/module_utils/wafv2.py:29]: I hope the two function become obsolete, once `wafv2` will get support for paginator - one day.

File: plugins/module_utils/wafv2.py
Comment (goneri) [inline plugins/module_utils/wafv2.py:48]: May I **suggest** to add a unit-test for this function?

File: plugins/module_utils/wafv2.py
Comment (goneri) [inline plugins/module_utils/wafv2.py:70]: Well, unit-test would be great for the others too :-).

File: plugins/module_utils/wafv2.py
Comment (markuman) [reply plugins/module_utils/wafv2.py]: Nope :) fixed it

**Review by goneri — APPROVED** [review]
Thank you @markuman.

File: plugins/module_utils/wafv2.py
Comment (goneri) [reply plugins/module_utils/wafv2.py:48]: Yes I know and this is the reason why I put `suggest` in bold. PLEASE, Don't force yourself to do it!

I see two reasons to write unit-tests in your case:

- sometime the behavior of the API change, it's a way to have a copy of the original expect input.
- unit-tests run way way way way faster than integration tests. And if they fail we can configure the CI to avoid the functional tests. This way we save the valuable developer time and a lot of CI resources.
- it's also rather easy for a developer to be able to run the unit-tests locally. It's a different story for the functional tests.

**Review by gravesm — APPROVED** [review]

File: plugins/module_utils/wafv2.py
Comment (tremble) [reply plugins/module_utils/wafv2.py:48]: For what it's worth you should be able to copy most of the unit-test framework for these transforms from https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/module_utils/core/test_scrub_none_parameters.py

**Review by jillr — APPROVED** [review]

**Pull request merged**
